project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php

```diff
@@ -4,129 +4,6262 @@
+            $i = 0;
+
+            $this->addParameterTranslations($parameterTranslations, t('Pressure', [], 'dataFixtures', $locale), t('15 bar', [], 'dataFixtures', $locale), $locale, $i);
+            $this->addParameterTranslations($parameterTranslations, t('Water reservoir capacity', [], 'dataFixtures', $locale), t('2 l', [], 'dataFixtures', $locale), $locale, $i);
+            $this->addParameterTranslations($parameterTranslations, t('Milk reservoir capacity', [], 'dataFixtures', $locale), t('600 ml', [], 'dataFixtures', $locale), $locale, $i);
+            $this->addParameterTranslations($parameterTranslations, t('Magazine capacity for beans', [], 'dataFixtures', $locale), t('400 g', [], 'dataFixtures', $locale), $locale, $i);
+        }
+
+        $this->setParametersByTranslations($productData, $parameterTranslations);
+
+        $this->setPriceForAllPricingGroups($productData, '19743.6');
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setSellingFrom($productData, '20.1.2014');
+        $this->setSellingTo($productData, null);
+        $productData->usingStock = true;
+        $productData->stockQuantity = 140;
+        $productData->outOfStockAction = Product::OUT_OF_STOCK_ACTION_HIDE;
+
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setAvailability($productData, AvailabilityDataFixture::AVAILABILITY_IN_STOCK);
+        $this->setCategoriesForAllDomains($productData, [CategoryDataFixture::CATEGORY_COFFEE]);
+        $this->setFlags($productData, [FlagDataFixture::FLAG_NEW_PRODUCT]);
+
+        $productData->sellingDenied = false;
+        $this->setBrand($productData, BrandDataFixture::BRAND_DELONGHI);
+
+        $this->createProduct($productData);
+
+        $productData = $this->productDataFactory->create();
+
+        $productData->catnum = '5962199';
+        $productData->partno = '';
+        $productData->ean = '8845781245958';
+
+        foreach ($this->domain->getAllLocales() as $locale) {
+            $productData->name[$locale] = t('Pot holder, black', [], 'dataFixtures', $locale);
+        }
+
+        foreach ($this->domain->getAllIncludingDomainConfigsWithoutDataCreated() as $domain) {
+            $productData->descriptions[$domain->getId()] = t('This pot holder is used to hold pots. No more burnt kitchen tables!', [], 'dataFixtures', $domain->getLocale());
+            $productData->shortDescriptions[$domain->getId()] = t('This pot holder is used to hold pots. No more burnt kitchen tables!', [], 'dataFixtures', $domain->getLocale());
+        }
+
+        $this->setPriceForAllPricingGroups($productData, '3');
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setSellingFrom($productData, '13.2.2014');
+        $this->setSellingTo($productData, null);
+        $productData->usingStock = false;
+        $productData->stockQuantity = 0;
+        $productData->outOfStockAction = Product::OUT_OF_
```

Comment:
we could discuss this further but honestly, I do not see any approach that could be much better (maybe we could move the values to a data source like CSV? :thinking: nah, just kidding :laughing: ) - you see, if there was a method that would accept all the data as parameters or accepted an associative array than it would probably worsen the readability. It is a spaghetti code but it serves it's purpose well IMHO. We could think about some further refactorings (I am even open to moving *some* values back to CSV) but I think this is not in the scope of this PR.